src/osemgrep/networking/Scan_helper.ml

```diff
@@ -14,16 +14,29 @@ let start_scan ~dry_run ~token url meta =
     let body = JSON.(string_of_json (Object [ ("meta", meta) ])) in
     match Http_helpers.post ~body ~headers scan_endpoint with
     | Ok body -> (
-        let json = JSON.json_of_string body in
-        match json with
-        | Object xs -> (
-            match List.assoc_opt "scan" xs with
-            | Some (Object dd) -> (
-                match List.assoc_opt "id" dd with
-                | Some (Int i) -> Ok (Some (string_of_int i))
-                | _else -> Ok None)
-            | _else -> Ok None)
-        | _else -> Ok None)
+        try
+          let json = JSON.json_of_string body in
+          match json with
```

Comment:
same here, I think this could be moved in an helper function, like xxx_of_json.
(and should also be specified in ATD maybe, so a TODO comment could be useful).

Reply:
done in [93bafdf](https://github.com/returntocorp/semgrep/pull/8149/commits/93bafdfd8882502b01454b97d93e2009c352ef8d)